src/tests/dictionaryConcepts/container/SelectAnswers.test.js

```diff
@@ -13,22 +17,59 @@ describe('<SelectAnswers />', () => {
       source: 'test source',
       frontEndUniqueKey: 'unique',
     };
-    wrapper = shallow(<SelectAnswers {...props} />);
+    wrapper = mount(<SelectAnswers {...props} />);
   });
 
-  it('should handle change in AsncSelect', () => {
-    const instance = wrapper.instance();
-    const spy = jest.spyOn(instance, 'handleChange');
-    instance.handleChange(['value']);
+  it('should handle key down event when a user presses enter button to search for concept', async () => {
+    const spy = jest.spyOn(wrapper.instance(), 'handleKeyPress');
+    const event = { keyCode: KEY_CODE_FOR_ENTER };
+    await wrapper.instance().handleKeyPress(event, 'malaria');
     expect(spy).toHaveBeenCalled();
-    instance.handleChange(null);
+  });
+
+  it('should handle key down event when a user presses enter button to search for concept with input less that 3', async () => {
```

Comment:
Do we allow searching with less than 3 characters?

Reply:
I have updated it to run a test on that the value for the state variable `isVisible` if false when one enters less than 3 characters since when one enters values that are less than 3 this is set to false. This can be seen [here](https://github.com/openmrs/openmrs-ocl-client/pull/387/files#diff-f894a03829d3d9ceb18310a3c2cc7179R29)

Reply:
Do you have a corresponding test which expects the opposite when then number of characters is not less than three?

Reply:
Yes I do the test is on line 23 [here](https://github.com/openmrs/openmrs-ocl-client/pull/387/files#diff-c2da5db27328a227e2b237bdd35abdd7R23)

Reply:
What difference does it make if both just say "should handle key down event..."? Shouldn't one expect the opposite of the other?

Reply:
They both handle the key down event that's why they both have "should handle key down event..." but they handle the key down event differently which is said by the other part of the description.

Reply:
If one expect(wrapper.state().isVisible).toBeFalsy(), shouldn't the other expect the opposite?

Reply:
I have updated the test for the case where the input is valid to check-expect the value of isVisible if its true `expect(wrapper.state().isVisible).toBeTruthy()`. this is also shown [on line 28 ](https://github.com/openmrs/openmrs-ocl-client/pull/387/files#diff-c2da5db27328a227e2b237bdd35abdd7R28)

Reply:
The tests say should handle key down. What is the difference? The test description does not seem to reflect it. Are you testing for handle key down for one but not the other test?

Reply:
The tests call the `handleKeyDown` function which handles the key down event hence the description.